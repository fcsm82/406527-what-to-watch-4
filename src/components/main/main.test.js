import React from "react";
import renderer from "react-test-renderer";
import Main from "./main.jsx";

const TOP_FILM = {
  title: `The Grand Budapest Hotel`,
  poster: `bg-the-grand-budapest-hotel.jpg`,
  genre: `Drama`,
  releaseDate: 2014
};

const films = [
  {
    id: `lKFDHkhaeud`,
    background: `bg-the-grand-budapest-hotel.jpg`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Drama`,
    releaseDate: 2020,
    poster: `the-grand-budapest-hotel-poster.jpg`,
    preview: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    rating: 8.9,
    level: `Very Good`,
    totalRatings: 240,
    director: `Wes Andreson`,
    starring: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`
    ]
  }
];

it(`Render Main`, () => {
  const tree = renderer
    .create(<Main
      TOP_FILM = {TOP_FILM}
      films = {films}
      onTitleClick = {() => {}}
      onPosterClick = {() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
