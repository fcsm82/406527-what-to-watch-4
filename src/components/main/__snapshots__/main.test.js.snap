// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Main 1`] = `
Array [
  <div
    className="visually-hidden"
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <symbol
        viewBox="0 0 19 20"
      >
        <title>
          +
        </title>
        <g
          fill="none"
          fillRule="evenodd"
          strokeWidth="1"
        >
          <polygon
            fill="#EEE5B5"
            points="10.778 11.288 10.778 19.553 8.4165 19.553 8.4165 11.288 0.62793 11.288 0.62793 8.9268 8.4165 8.9268 8.4165 0.66211 10.778 0.66211 10.778 8.9268 18.566 8.9268 18.566 11.288"
          />
        </g>
      </symbol>
      <symbol
        fill="#FFF9D9"
        viewBox="0 0 27 27"
      >
        <path
          clipRule="evenodd"
          d="M23.8571 0H16V3.14286H23.8571V11H27V3.14286V0H23.8571Z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M27 23.8571V16H23.8571V23.8571H16V27H23.8571H27L27 23.8571Z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="m0 3.1429v7.8571h3.1429v-7.8571h7.8571v-3.1429h-11v3.1429z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M3.14286 27H11V23.8571H3.14286L3.14286 16H0L0 23.8571V27H3.14286Z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        viewBox="0 0 18 14"
      >
        <path
          clipRule="evenodd"
          d="M2.40513 5.35353L6.1818 8.90902L15.5807 0L18 2.80485L6.18935 14L0 8.17346L2.40513 5.35353Z"
          fill="#EEE5B5"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        viewBox="0 0 14 21"
      >
        <g
          fill="#EEE5B5"
          fillRule="evenodd"
          strokeWidth="1"
        >
          <polygon
            fillRule="nonzero"
            points="0 -1.1191e-13 4 -1.1191e-13 4 21 0 21"
          />
          <polygon
            fillRule="nonzero"
            points="10 -1.1191e-13 14 -1.1191e-13 14 21 10 21"
          />
        </g>
      </symbol>
      <symbol
        id="play-s"
        viewBox="0 0 19 19"
      >
        <path
          clipRule="evenodd"
          d="M0 0L19 9.5L0 19V0Z"
          fill="#EEE5B5"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>,
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel"
            height="327"
            src="img\\\\the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Comedies
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Documentary
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Dramas
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Horror
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Kids & Family
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Romance
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Sci-Fi
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Thrillers
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Fantastic Beasts: The Crimes of Grindelwald"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
